processor/src/main/kotlin/com/thinkslynk/fabric/annotations/extensions/TypeSpec.kt

```diff
@@ -2,37 +2,33 @@ package com.thinkslynk.fabric.annotations.extensions
 
 import com.squareup.kotlinpoet.*
 import javax.lang.model.element.Element
-import javax.lang.model.element.ElementKind
-import javax.lang.model.element.ExecutableElement
 
 fun TypeSpec.Builder.addDefaultProperties(
     elements: Collection<Element>,
     formatName: (name: String) -> String
 ): TypeSpec.Builder {
-    var b = this
-    elements.forEach {
+    elements.forEach {element->
 
-        val targetClassName = it.simpleName.toString()
+        val targetClassName = element.simpleName.toString()
         val propName = formatName(targetClassName)
-        if (it.kind == ElementKind.METHOD) {
-            val methodElement = it as ExecutableElement
-            val returnType = methodElement.returnType
+        if (element.isMethod()) {
+            val returnType = element.returnType
 
-            b = b.addProperty(
+            this.addProperty(
```

Comment:
I'm sure that they are mutated as I've used them before and rely on it in other places of my code. Although I couldn't find a mention of them in the docs.
AFAIK almost all builders in java are mutable as that's their main point for existing. For building immutable objects.

Reply:
Maybe it's common in java, not sure. In python I use `dataframe`s all the time, and those do not mutate state. I'd figure with kotlin's data classes, it would be common to mutate state with a `.copy(field=newVal)` so that you build through immutable objects. 

But point taken that you've tested this and it's fine. Just more wanted to explain why I did it the other way to begin with. If it functions properly this way, I'm all for it.